### DISCUSSION TITLE
Durchsuchen von Unterordnern / searching of subfolders

### DISCUSSION THREAD

**Discussion by BeckeBauer** [geimist/synOCR#41] [Q&A] [answered]
Finde das Projekt eine großartige Idee. Ich habe eine Menge pdf's, die bereits fein säuberlich in Unterordner sortiert sind. Wenn ich diese durchsuchbar machen möchte, müsste ich sie dann alle "rausholen" und in Verzeichnis legen bzw. für jedes der (wirklichen vielen) Unterordner einen eigenen Pfadeintrag anlegen? Oder lässt sich irgendwo einstellen, dass auch Unterordner des angegebenen Pfads einbezogen werden?

Great project, thank you. I have a lot of pdf's which are sorted into subfolders. When I want them to be searchable, do I need to extract them from their current location and put them into one folder or do I need to configure for each of my (many many) subfolders an own path? Or can I set an option to include subfolders of the configured path?

**Comment by geimist** [comment]
Hi,
synOCR ist für den kontinuierlichen Workflow in Verbindung mit einem Scanner konzipiert. In diesem Szenario wird auch nur ein Ordner benötigt. 
Um eine bestehende Struktur zu 'OCRen' habe ich ein kleines Hilfskript bereitgestellt: [FAQ](https://github.com/geimist/synOCR/wiki/04_FAQ-(de)#%EF%B8%8F-wie-kann-ich-eine-bestehende-ordnerstruktur-%C3%BCbernehmen).
Schau dir das mal bitte an. Sollten noch Fragen auftreten, einfach melden.

LG
Stephan

**Reply by BeckeBauer** [reply]
Super - schaue ich mir an. Vielen Dank

**Reply by BeckeBauer** [reply]
Hi Stephan, 
habe gerade das Script gestartet. Es verschiebt alle pdfs in den angegebenen synOCR-INPUT-Ordner mit folgender Meldung:

```
root@dc:/volume1/test/#bash recursive_inputdir_workflow.sh
touch: cannot touch 'recursive_inputdir_workflow.sh/multidir_workflow_INDEX.txt' : Not a directory
Index wird erstellt ➜ verschiebe Dateien in den Arbeitsordner
recursive_inputdir_workflow.sh: line 56: recursive_inputdir_workflow.sh/multidir_workflow_INDEX.txt: Not a directory
recursive_inputdir_workflow.sh: line 56: recursive_inputdir_workflow.sh/multidir_workflow_INDEX.txt: Not a directory
.... (die Zeile "recursive...." wiederholt sich noch viele Male)
```
Ich habe eine Weile gewartet, aber im OUTPUT-Ordner erscheinen keine pdfs (das hätte ich vermutet, wenn alles korrekt gelaufen wäre). Ein erneuter Aufruf des Befehls wiederholt nur die Ausgabe 
```
touch: cannot touch 'recursive_inputdir_workflow.sh/multidir_workflow_INDEX.txt' : Not a directory
Index wird erstellt ➜ verschiebe Dateien in den Arbeitsordner
```
Was mache ich falsch?

**Reply by geimist** [reply]
Hast du die Verzeichnisse im Skript angepasst?
Verwendest du die vollständigen Pfade (beginnen mit `/volume...`)? 

Probiere am besten auch mit einen Testverzeichnis.

**Reply by BeckeBauer** [reply]
Ich denke schon, denn alle pdfs aus der Struktur sind in das im Script definierte Input-Verzeichnis verschoben worden. Es passiert dann nur nichts mehr; im output Ordner erscheint nicht eine einzige (durchsuchbare) pdf oder bin ich zu ungeduldig?

**Reply by BeckeBauer** [reply]
Beim weiteren rumprobieren habe ich festgestellt, dass ich das script nicht mit `!/bin/bash` und dann mit `./recursive_inputdir_workflow.sh` gestartet habe. Dies habe ich jetzt nachgeholt und das Verschieben in den Input-Ordner hat ohne Fehlermeldung geklappt (im Script-Ordner ist jetzt die `multidir_workflow_INDEX.txt` aufgetaucht (kenne mich leider nicht mit Skripten aus, daher habe ich den das Scrip zuvor mit `bash recursive_inputdir_workflow.sh` gestartet). Jetzt muss ich warten oder kann ich irgendwie erkennen, dass die Texterkennung arbeitet?

**Comment by geimist** [comment]
Schön, dass es jetzt geht. Ja, in deinem Fall wurde der eigene Pfad nicht erkannt. Dass deine Art des Aufrufs Einfluss darauf hat, ist mir auch neu. Ich werde da mal eine Prüfung mit einbauen.

> … Jetzt muss ich warten oder kann ich irgendwie erkennen, dass die Texterkennung arbeitet?
Du musst wie beschrieben synOCR starten (sofern noch nicht geschehen) und auf den Hinweis achten:
"synOCR-Profil ohne Umbenennungssyntax und Einsortierung in regeldefinierte Ordner aufrufen, d. h. alle fertigen PDFs liegen im synOCR OUTPUT-Ordner".

Ob synOCR läuft, siehst du an der Systemauslastung und z. B. an dem Container synOCR im Container Manager.

**Reply by BeckeBauer** [reply]
Leider passiert nichts. Kann ich den OCR-Prozess manuell anstoßen?

**Reply by geimist** [reply, marked as answer]
Dieses Skript ruft **_NICHT_** synOCR auf. Wenn du keine Ordnerüberwachung oder einen Zeitplan nutzt, dann musst du es manuell starten.
Bitte beachte die Hinweise bzgl. Umbenennung und Zielverzeichnis.

**Comment by BeckeBauer** [comment]
> Dieses Skript ruft **_NICHT_** synOCR auf. Wenn du keine Ordnerüberwachung oder einen Zeitplan nutzt, dann musst du es manuell starten.

Sorry, war naheliegend (ich bin ein Depp). Vielleicht ein entsprechender Hinweis im Script für Leute wie mich wäre supi...

> Bitte beachte die Hinweise bzgl. Umbenennung und Zielverzeichnis.

Ich habe an den Voreinstellungen in synORC nichts geändert (abgesehen vom Pfad zu den pdfs). Ich hoffe, die Datumseinstellungen funkten nicht dazwischen (ich konnte da nichts deaktivieren). Mal sehen, was passiert...

**Reply by geimist** [reply]
Dann sollte es passen. Wichtig ist, dass `§tit` (die Variable für den ursprünglichen Dateinamen) in der Umbenennungssyntax enthalten ist. In den Defaulteinstellungen ist das der Fall.

**Reply by BeckeBauer** [reply]
Die Voreinstellungen fügen - wie ich feststellen musste - ein Datum vor den Dateinamen. Zum Glück gibt es bulk-rename Programme - insofern war das kein Problem, diese in einem Rutsch wieder von allen pdfs zu entfernen. Dann mussten noch die nicht "OCR-baren" pdfs händisch in den Output-Ordner kopiert werden und dann das Scrip angestoßen werden (da ich zu ungeduldig war, musste ich noch das log-file anpassen, d.h. die Endung wieder entfernen). Und dann sind die pdfs fein säuberlich zurücksortiert worden 👍

**Reply by geimist** [reply]
Der Originaldateiname wird mit gesichert und beim Zurückschieben wiederhergestellt. Du hättest das also nicht händisch machen müssen.

**Reply by BeckeBauer** [reply]
Wieder was gelernt - beim nächsten Mal geht alles noch viel besser...
